proto/api.proto

```diff
@@ -84,6 +104,9 @@ message RequestHeader {
   // that the operation takes place on the key range from Key to EndKey,
   // including Key and excluding EndKey.
   optional bytes end_key = 4 [(gogoproto.nullable) = false, (gogoproto.customtype) = "Key"];
+  // A key for a request referencing data in a Table. A request referencing
+  // a table must have an empty key and end_key above.
+  optional TableKey table_key = 11 [(gogoproto.nullable) = false];
```

Comment:
Like `key` and `end_key`, I think we're going to want separate `table_key` and `table_end_key`. Though thinking about this more, it seems unlikely that we'd ever want a request to span tables. So perhaps pull `table_name` and `index_name` out of `TableKey` and into `RequestHeader`.